### PR TITLE
Fix Series.unstack to support non-numeric type and keep the names

### PR DESCRIPTION
**Pull request by itholic**
According to https://github.com/databricks/koalas/pull/1501#discussion_r426985880 and https://github.com/databricks/koalas/pull/1501#issuecomment-631070677, Fix `Series.unstack()` to support non-numeric type and keep the names of index and columns.

```python
>>> kser = ks.Series(list('abcd'), index=pd.MultiIndex.from_product([['one', 'two'], ['a', 'b']], names=["A", "B"]))
>>> kser
A    B
one  a    a
     b    b
two  a    c
     b    d
Name: 0, dtype: object

>>> kser.unstack()
B    a  b
A
two  c  d
one  a  b
```

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -4862,12 +4862,19 @@ def unstack(self, level=-1):
...
         internal = InternalFrame(
             spark_frame=sdf,
-            index_map=OrderedDict((index_scol_name, None) for index_scol_name in index_scol_names),
+            index_map=OrderedDict(
+                (index_scol_name, index_map[index_scol_name])
+                for index_scol_name in index_scol_names
+            ),
+            column_label_names=[index_map[pivot_col][0]]
+            if index_map[pivot_col] is not None
+            else None,
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:4877]: Can you make separate variables for both `index_map` and `column_label_names`?

**Comment by HyukjinKwon** [comment]
LGTM otherwise.

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -4862,12 +4862,19 @@ def unstack(self, level=-1):
...
-        sdf = sdf.groupby(index_scol_names).pivot(pivot_col).sum(data_scol_name)
+        sdf = sdf.groupby(index_scol_names).pivot(pivot_col).agg(F.first(scol))
         internal = InternalFrame(
             spark_frame=sdf,
-            index_map=OrderedDict((index_scol_name, None) for index_scol_name in index_scol_names),
+            index_map=OrderedDict(
+                (index_scol_name, index_map[index_scol_name])
+                for index_scol_name in index_scol_names
+            ),
+            column_label_names=[index_map[pivot_col][0]]
```
Comment (ueshin) [inline databricks/koalas/series.py:4875]: `name_like_string(index_map[pivot_col])` instead of `index_map[pivot_col][0]`?

Comment (itholic) [reply databricks/koalas/series.py]: Thanks, @ueshin !

File: databricks/koalas/series.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/series.py]: Thanks, @HyukjinKwon !

**Pull request merged**